substratevm/src/com.oracle.svm.core/src/com/oracle/svm/core/hub/DynamicHub.java

```diff
@@ -266,6 +266,12 @@
      */
     private ClassInitializationInfo classInitializationInfo;
 
+    /**
+     * Indicates if this class originally has a <clinit> method. It is used for serialization
+     * support.
+     */
+    private boolean hasCLinit;
```

Comment:
I think this information can be easily merged into `ClassInitializationInfo` above by creating another constant like `INITIALIZED_INFO_SINGLETON`, for example `NO_INITIALIZER_INFO_SINGLETON`. This way, not every `DynamicHub` needs to have this extra field.

Reply:
fixed